dcm2bids/sidecar.py

```diff
@@ -35,17 +35,20 @@ def __lt__(self, other):
         lts = []
         for key in self.compKeys:
             try:
-                lts.append(self.data.get(key) < other.data.get(key))
+                if all(key in d for d in (self.data, other.data)):
+                    if self.data.get(key) == other.data.get(key):
+                        lts.append(None)
+                    else:
+                        lts.append(self.data.get(key) < other.data.get(key))
+                else:
+                    lts.append(None)
+
             except:
-                lts.append(False)
+                lts.append(None)
 
         for lt in lts:
-            if lt:
-                return True
-            else:
-                pass
-
-        return False
+            if lt is not None:
+                return lt
```

Comment:
Why was fixing this needed for this PR at all? The test you added doesn't use `<`.

Reply:
Again, the order of the keys you use to convert matters. I agree it should have been in an other PR but this was bugging me especially because within the tests (already coded) the order was not correct with the expected behavior. Not this is tested and validated.

Reply:
```
def __lt__(self, other):
    compKeys = [key for key in self.compKeys if key in self.data and key in other.data]
    return [self.data[key] for key in compKeys] < [self.other[key] for key in compKeys]
```

respects the order in `compKeys`, because the list-comprehensions are ordered.